scripts/test/SANS/gui_logic/test_run_tab_presenter.py

```diff
@@ -629,65 +632,6 @@ def test_that_default_name_is_used_when_export_table(self):
         args = [arg for arg in self.presenter.display_save_file_box.call_args[0]]
         self.assertEqual(args, expected_args)
 
-    def test_that_canSAS_is_disabled_if_2D_reduction(self):
```

Comment:
I'm trying to understand more about the changes you've implemented. From seeing that this test has been removed and from playing with the radio buttons and checkboxes it seems that you can now use canSAS even in 2D reduction. Is that ok? I admit I did try to run it and whilst  I got some files listed in Workspaces nothing saved in the folder and I got an error message. Would this error message have been prevented previously by the design that disabled canSAS in 2d reduction?

Reply:
So prior to this PR the radio buttons incorrectly always used the 2D save method (NXSCanSas) and did not automatically switch as the logic was not firing in certain cases, as it was being done in a mix of the view and presenter.

Now that the logic is correctly in the model and (checking with the SANS group) the default option which automatically toggles "should be" the they save in. As RKH is a legacy format they're trying to move away from.
Because of this I initially simplified the logic on our end, as users who are fiddling with the save options are likely instrument scientists enabling this format despite it gradually going away.

As per the bug you found above where it gets greyed out, I think I'll remove the step greying it out as it becomes overly complex as the correct approach. For example, do we untick CanSAS and grey it out whilst someone switches to 2D mode, or leave it ticked and greyed out? Do we sometimes ignore the state in our model and do something different? What happens when we switch back to 1D mode, do we restore the tick box?

Because of all of this, I decided to leave it as "what you see is what you get". As long as it gives a sane error message (which I can check) I'm okay with this as a compromise, as a user has to manually adjust the saving options to have this happen, so takes responsibility at that point.

However, again I'm open to thoughts and opinions on this?

Reply:
I think it is probably a good idea to remove the greying out as that can potentially lead to more confusion. So long as it switches to the correct tick box depending on the resolution choice then perhaps it's fair that manually changing the save type is at the user's discretion.

Could you check the error message though? I think it mentioned something about histograms but to a non-expert user like myself I'm not sure it was particularly clear it was a bad choice of save type.